### PR TITLE
Faster dedup v1.8

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

port #22607

#### Summary of Changes

Fixes #

### CODE REVIEW

**Review by jstarry — APPROVED** [review]

File: perf/src/sigverify.rs [hunk 1]
```diff
@@ -390,35 +393,80 @@ pub fn generate_offsets(
...
+    filter: Vec<AtomicU64>,
+    seed: RandomState,
+    age: Instant,
+    max_age: Duration,
+    saturated: AtomicBool,
+}
+
+impl Deduper {
+    pub fn new(size: u32, max_age_ms: u64) -> Self {
+        let mut filter: Vec<AtomicU64> = Vec::with_capacity(size as usize);
```
Comment (sakridge) [inline perf/src/sigverify.rs:406]: same as the vec! macro right?
`= vec![size; Default::default());`

File: perf/src/sigverify.rs [hunk 2]
```diff
@@ -21,7 +20,11 @@ use {
         hash::Hash, message::MESSAGE_HEADER_LENGTH, pubkey::Pubkey, short_vec::decode_shortu16_len,
         signature::Signature,
     },
-    std::sync::atomic::{AtomicU64, Ordering},
+    std::collections::hash_map::RandomState,
+    std::hash::BuildHasher,
+    std::hash::Hasher,
```
Comment (sakridge) [inline perf/src/sigverify.rs:25]: I think you're getting siphasher here actually. wouldn't it be better to specify ahash explicitly so we know what we're getting?

Comment (aeyakovenko) [reply perf/src/sigverify.rs:25]: switched to ahash

File: perf/src/sigverify.rs (hunk 1 above)
Comment (aeyakovenko) [reply perf/src/sigverify.rs:406]: |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `AtomicU64`

Comment (aeyakovenko) [reply perf/src/sigverify.rs:406]: doesn't like it

Comment (sakridge) [reply perf/src/sigverify.rs:406]: ah, ok

**Pull request merged**